test/test_polymorphic_value.cpp

```diff
@@ -125,12 +125,12 @@ TEST_F(PolymorphicValueTest, Struct) {
     PolymorphicValue b = type.create();
     b->*"x" = 2788;
     b->*"y" = 2.71828;
-    EXPECT_EQ((PolymorphicValue)(b->*"x"), 2788);
-    EXPECT_EQ((PolymorphicValue)(b->*"y"), 2.71828);
+    EXPECT_EQ(b->*"x", PolymorphicValue(2788));
```

Comment:
Why previously RHS was just `2788` but now it needs to be `PolymorphicValue(2788)`?

Reply:
This is a limitation of C++. In order to use operators of `PolymorphicValue`, one of the parameter must be `PolymorphicValue` otherwise, there would be conflicts, for example, if you have `1L + 1.0`, then you can cast both to `double` and use the `operator+` for `double`, you can also cast both to `PolymorphicValue` and use the `operator+` of `PolymorphicValue`, and C++ can not tell which to use.